testlib/src/main/java/io/spine/testing/ClassTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2020, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.testing;
+
+import com.google.common.testing.NullPointerTester;
+import com.google.common.testing.NullPointerTester.Visibility;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Modifier;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.testing.NullPointerTester.Visibility.PUBLIC;
+import static io.spine.testing.Tests.assertTrue;
+
+/**
+ * Abstract base for test suites that test a class (e.g. static methods) rather than an object.
+ *
+ * @param <C>
+ *         the class under the tests
+ */
+@SuppressWarnings("UnstableApiUsage") // NPE testing utils from Guava.
+public abstract class ClassTest<C> {
+
+    private final Class<C> subject;
```

Comment:
The `Class` itself is not a subject per se, rather the instance and/or static methods are.

Reply:
I suggest renaming.

Reply:
When we test a utility class, it's the class that's under consideration. This is what in the language we speak. When it comes to singleton, it's an class which produces only one object. Also, the name was given in line with what we have in Google Truth.

Reply:
@armiol, I've requested your review hoping you may suggest a better name for class under tests.

Reply:
@alexander-yevsyukov @dmdashenkov I was ready to suggest something like `StaticApiTest`. While it's a bad name in general, it defines the purpose of derived tests more or less precisely. 
But now, as I see, we also test the class-under-testing for being `final` and for having private and parameter-less ctor, I don't think it's a good name either.

`ClassTest` feels a bit generic to me. By its name, it should be applicable to testing any class, while it's not quite.

So let me think a couple of dozen minutes more.

Reply:
@alexander-yevsyukov @dmdashenkov as to `subject`, this variable name looks fine to me. It's tightly related to the name of the `ClassTest`. So, if we keep `ClassTest`, we should keep `subject` for sure, because that's what we test.

Reply:
@alexander-yevsyukov @dmdashenkov to satisfy each and every, could we use `private final Class<C> input;`?